File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    pokemon = {}
    
    for num in nums:
        if num in pokemon:
            pokemon[num] += 1
        else:
            pokemon[num] = 1
            
    if len(nums) // 2 <= len(pokemon):
        return len(nums) // 2
    else:
        return len(pokemon)

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    const N = nums.length;
    const pokemon = {};
    
    for (const num of nums) {
        if (!pokemon[num]) {
            pokemon[num] = 1
        } else {
            pokemon[num] += 1
        }
    }
    
    if (Object.keys(pokemon).length >= N / 2) {
        return N / 2;
    } else {
        return Object.keys(pokemon).length;
    }
}